lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1256,9 +1282,8 @@ private function getSQLForJoins(string $fromAlias, array &$knownAliases) : strin
     {
         $sql = '';
 
-        if (isset($this->sqlParts['join'][$fromAlias])) {
-            /** @var Join $join */
-            foreach ($this->sqlParts['join'][$fromAlias] as $join) {
+        if (isset($this->join[$fromAlias])) {
```

Comment:
Could an eary return be used here?

Reply:
Sure. 81d8c800b5fe36610ee5c735b803d72ed1a405da.